Vates/ParaviewPlugins/ParaViewReaders/MDHWNexusReader/vtkMDHWNexusReader.cxx

```diff
@@ -158,36 +149,33 @@ int vtkMDHWNexusReader::RequestData(vtkInformation * vtkNotUsed(request), vtkInf
 }
 
 int vtkMDHWNexusReader::RequestInformation(
-  vtkInformation *vtkNotUsed(request),
-  vtkInformationVector **vtkNotUsed(inputVector),
-  vtkInformationVector *outputVector)
-{
-  if(m_presenter == NULL)
-  {
-    m_presenter = new MDHWNexusLoadingPresenter(new MDLoadingViewAdapter<vtkMDHWNexusReader>(this), FileName);
+    vtkInformation *vtkNotUsed(request),
+    vtkInformationVector **vtkNotUsed(inputVector),
+    vtkInformationVector *outputVector) {
+  if (m_presenter == nullptr) {
+    std::unique_ptr<MDLoadingView> view =
+        Mantid::Kernel::make_unique<MDLoadingViewAdapter<vtkMDHWNexusReader>>(
+            this);
+    m_presenter = Mantid::Kernel::make_unique<MDHWNexusLoadingPresenter>(
+        std::move(view), FileName);
   }
 
-  if (m_presenter == NULL)
-  {
-    // updater information has been called prematurely. We will reexecute once all attributes are setup.
+  if (m_presenter == nullptr) {
+    // updater information has been called prematurely. We will reexecute once
+    // all attributes are setup.
     return 1;
   }
-  if(!m_presenter->canReadFile())
-  {
-    vtkErrorMacro(<<"Cannot fetch the specified workspace from Mantid ADS.");
+  if (!m_presenter->canReadFile()) {
+    vtkErrorMacro(<< "Cannot fetch the specified workspace from Mantid ADS.");
     return 0;
   }
-  
+
   m_presenter->executeLoadMetadata();
   setTimeRange(outputVector);
-  MDHWNexusLoadingPresenter *castPresenter =
-      dynamic_cast<MDHWNexusLoadingPresenter *>(m_presenter);
-  if (castPresenter) {
-    std::vector<int> extents = castPresenter->getExtents();
-    outputVector->GetInformationObject(0)
-        ->Set(vtkStreamingDemandDrivenPipeline::WHOLE_EXTENT(), &extents[0],
-              static_cast<int>(extents.size()));
-  }
+  std::vector<int> extents = m_presenter->getExtents();
+  outputVector->GetInformationObject(0)
+      ->Set(vtkStreamingDemandDrivenPipeline::WHOLE_EXTENT(), &extents[0],
+            static_cast<int>(extents.size()));
```

Comment:
I am not sure I understand, why is the `if` gone, or rather, why can we no do without the cast?